shutter.py

```diff
@@ -0,0 +1,196 @@
+
+    category = CATEGORY_GARAGE_DOOR_OPENER
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.add_preload_service('GarageDoorOpener')\
+            .configure_char(
+                'TargetDoorState', setter_callback=self.change_state)
+
+    def change_state(self, value):
+        logging.info("GDC value: %s", value)
+        self.get_service('GarageDoorOpener')\
+            .get_characteristic('CurrentDoorState')\
+            .set_value(value)
+
+
+class WindowCovering(Accessory):
+    """Fake WindowCovering."""
+
+    category = CATEGORY_WINDOW_COVERING
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        """Quick assign basic accessory information."""
+        self.set_info_service(firmware_revision=2,  manufacturer="Brand",
+                model="Shutter", serial_number="0123456789")
+        
+        # Add the WindowCovering service. Also add optional characteristics to it.
+        serv_cover = self.add_preload_service(
+            'WindowCovering', chars=['ObstructionDetected'])
+
+        self.char_target_pos = serv_cover.configure_char(
+            'TargetPosition', setter_callback=self.set_target_position)
+            
+        self.char_state = serv_cover.configure_char(
+            'PositionState', setter_callback=self.set_position_state)
+            
+        self.char_cur_pos = serv_cover.configure_char(
+            'CurrentPosition', setter_callback=self.set_current_position)
+                                     
+        #serv_cover.configure_char('PositionState', value=2) # STOPPED
+        #serv_cover.configure_char('ObstructionDetected', value=1)
+                                     
+    def set_target_position(self, value):
+        logging.info("WindowCovering TargetPosition value: %s", value)
+        self.get_service('WindowCovering')\
+            .get_characteristic('TargetPosition')\
+            .set_value(value)
+
+        time.sleep(5)   # Delays for 5 seconds closing the shutter.
+
+        logging.info("WindowCovering CurrentPosition value: %s", value)
+        self.get_service('WindowCovering')\
+            .get_characteristic('CurrentPosition')\
+            .set_value(value)
+
+    # The value property of PositionState must be one of the following:
+    # Characteristic.PositionState.DECREASING = 0;
+    # Characteristic.PositionState.INCREASING = 1;
+    # Characteristic.PositionState.STOPPED = 2; 
+    def set_position_state(self, value):
+        logging.info("WindowCovering PositionState value: %s", value)
+        self.get_service('PositionState')\
+            .get_characteristic('PositionState')\
+            .set_value(value)
+
+    def set_current_position(self, value):
+        logging.info("WindowCovering CurrentPosition value: %s", value)
+        self.get_service('WindowCovering')\
+            .get_characteristic('CurrentPosition')\
+            .set_value(value)
```

Comment:
The methods `set_position_state` and `set_current_position` can be removed, since they are never called.